crates/autopilot/src/decoded_settlement.rs

```diff
@@ -231,76 +241,84 @@ impl DecodedSettlement {
                     flags: trade.8.into(),
                     executed_amount: trade.9,
                     signature: trade.10,
-                })
-                .collect(),
+                };
+                match trade.uid(domain_separator, &tokens) {
+                    Ok(order_uid) => Some(TradeWithOrderUid {
+                        inner: trade,
+                        order_uid,
+                    }),
+                    Err(err) => {
+                        tracing::error!(
+                            ?err,
+                            ?trade,
+                            "failed to calculate order uid, we don't know which order this trade \
+                             belongs to"
+                        );
+                        None
+                    }
+                }
+            })
+            .collect();
+        Ok(Self {
+            tokens,
+            clearing_prices,
+            trades,
             interactions: interactions.map(|inner| inner.into_iter().map(Into::into).collect()),
             metadata: metadata.try_into().ok().map(Bytes),
         })
     }
 
     /// Returns the total surplus denominated in the native asset for the
     /// solution.
-    pub fn total_surplus(&self, external_prices: &ExternalPrices) -> U256 {
-        self.trades.iter().fold(0.into(), |acc, trade| {
-            acc + match surplus(trade, &self.tokens, &self.clearing_prices, external_prices) {
-                Some(surplus) => surplus,
-                None => {
+    pub fn total_surplus(
+        &self,
+        external_prices: &ExternalPrices,
+        jit_order_uids: HashSet<OrderUid>,
```

Comment:
I think this should take the original auction instance which contains prices and all eligible order uids

Reply:
I will take care of this in a new autopilot domain implementation.